libminifi/include/properties/Decryptor.h

```diff
@@ -0,0 +1,55 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <string>
+
+#include "utils/EncryptionUtils.h"
+#include "utils/OptionalUtils.h"
+
+namespace org {
+namespace apache {
+namespace nifi {
+namespace minifi {
+
+class Configure;
+
+class Decryptor {
+ public:
+  explicit Decryptor(const utils::crypto::Bytes& encryption_key);
+  static bool isEncrypted(const utils::optional<std::string>& encryption_type);
+  std::string decrypt(const std::string& encrypted_text, const std::string& aad) const;
+  void decryptSensitiveProperties(Configure& configure) const;
+
+ private:
+  const utils::crypto::Bytes encryption_key_;
+};
+
+inline Decryptor::Decryptor(const utils::crypto::Bytes& encryption_key) : encryption_key_(encryption_key) {}
+
+inline bool Decryptor::isEncrypted(const utils::optional<std::string>& encryption_type) {
```

Comment:
I think this name is misleading, because the question it answer is not whether the argument is encrypted, but rather whether it's a valid encryption type string. I suggest calling it `isValidEncryptionType` or similar, move it near `utils::crypto::EncryptionType` and make it take a string rather than an optional string. (Move the optional validity check to the usage site)

later:
After looking at the test case, I realize that this is checking that it valid and it's not plaintext, so the encryption type is encrypted. I maintain that since it's checking a property of an encryption type string, it should be near the EncryptionType class. But not sure how to resolve the naming confusion, maybe you have a better idea?

Reply:
I have renamed it to `isValidEncryptionMarker`